libs/features/placeholders.js

```diff
@@ -1,30 +1,94 @@
-const placeholders = {};
-let fetched = false;
+const cachedPlaceholders = {};
 
 // eslint-disable-next-line no-async-promise-executor
-const fetchPlaceholders = (config) => new Promise(async (resolve) => {
-  if (fetched) { resolve(placeholders); return; }
+const fetchPlaceholders = (config, sheet) => new Promise(async (resolve) => {
+  if (cachedPlaceholders?.[config.locale.ietf]?.[sheet]) {
+    resolve(cachedPlaceholders[config.locale.ietf][sheet]);
+    return;
+  }
+
   const path = `${config.locale.contentRoot}/placeholders.json`;
-  const resp = await fetch(path);
+  const query = sheet !== 'default' && typeof sheet === 'string' && sheet.length ? `?sheet=${sheet}` : '';
+  const resp = await fetch(`${path}${query}`);
   const json = resp.ok ? await resp.json() : { data: [] };
-  if (json.data.length === 0) { resolve(placeholders); return; }
+  if (json.data.length === 0) { resolve({}); return; }
+  cachedPlaceholders[config.locale.ietf] = cachedPlaceholders[config.locale.ietf] || {};
+  cachedPlaceholders[config.locale.ietf][sheet] = {};
   json.data.forEach((item) => {
-    placeholders[item.key] = item.value;
+    cachedPlaceholders[config.locale.ietf][sheet][item.key] = item.value;
   });
-  fetched = true;
-  resolve(placeholders);
+  resolve(cachedPlaceholders[config.locale.ietf][sheet]);
 });
 
-function findPlaceholder(suppliedPlaceholders, key) {
-  return suppliedPlaceholders[key] || key.replaceAll('-', ' ');
+function keyToStr(key) {
+  return key.replaceAll('-', ' ');
 }
 
-export async function replaceText(config, regex, text) {
-  const fetchedPlaceholders = await fetchPlaceholders(config);
-  return text.replaceAll(regex, (_, key) => findPlaceholder(fetchedPlaceholders, key));
+async function getPlaceholder(key, config, sheet) {
+  const defaultLocale = 'en-US';
+
+  const getDefaultPlaceholders = async () => {
```

Comment:
I'd rename to getDefaultPlaceholder? I was kind of expecting it to return an array of placeholders otherwise.

Reply:
Nice catch, the method was returning either a set of placeholders or just one, which was misleading. The purpose of this method is to return a whole set, the name has been kept. However, other pieces of logic have been updated accordingly.